registry/storage/driver/testsuites/testsuites.go

```diff
@@ -1224,3 +1239,27 @@ func firstPart(filePath string) string {
 		filePath = dir
 	}
 }
+
+type errReader struct {
+	n   int64
+	msg string
+}
+
+func (er *errReader) Read(p []byte) (int, error) {
+	toread := int64(len(p))
```

Comment:
@RichardScothern , the purpose of this errReader is that we can read some bytes and return an error so that a storage driver can create the file with some contents and number of that bytes can be set when create the errReader.